src/security-council-mgmt/factories/L2SecurityCouncilMgmtFactory.sol

```diff
@@ -0,0 +1,252 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "../governors/SecurityCouncilMemberElectionGovernor.sol";
+import "../governors/SecurityCouncilNomineeElectionGovernor.sol";
+import "../SecurityCouncilManager.sol";
+import "../SecurityCouncilMemberRemovalGovernor.sol";
+import "@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+import "../interfaces/ISecurityCouncilManager.sol";
+import "../../ArbitrumTimelock.sol";
+import "@openzeppelin/contracts-upgradeable/governance/utils/IVotesUpgradeable.sol";
+
+struct DeployParams {
+    address _govChainEmergencySecurityCouncil;
+    address _l1CoreGovTimelock;
+    address _l2CoreGovTimelock;
+    address _proxyAdmin;
+    address[] _secondCohort;
+    address[] _firstCohort;
+    address l2UpgradeExecutor;
+    address arbToken;
+    uint256 _minL1TimelockDelay;
+    uint256 _removalGovVotingDelay;
+    uint256 _removalGovVotingPeriod;
+    uint256 _removalGovQuorumNumerator;
+    uint256 _removalGovProposalThreshold;
+    uint256 _removalGovVoteSuccessNumerator;
+    uint64 _removalGovMinPeriodAfterQuorum;
+    SecurityCouncilData[] _securityCouncils;
+    uint256 _fullWeightDuration;
+    // governor params
+    Cohort firstCohort;
+    SecurityCouncilNomineeElectionGovernor.Date firstNominationStartDate;
+    uint256 nominationFrequency;
+    uint256 nomineeVettingDuration;
+    address nomineeVetter;
+    uint256 nomineeQuorumNumerator;
+    uint256 nomineeVotingPeriod;
+    uint256 memberVotingPeriod;
+    uint256 memberFullWeightDurationNumerator;
+    uint256 memberDecreasingWeightDurationNumerator;
+    uint256 memberDurationDenominator;
```

Comment:
removed unused parameters, moved _fullWeightDuration to the bottom where duration numerators/denominator used to be

```suggestion
    // governor params
    Cohort firstCohort;
    SecurityCouncilNomineeElectionGovernor.Date firstNominationStartDate;
    uint256 nomineeVettingDuration;
    address nomineeVetter;
    uint256 nomineeQuorumNumerator;
    uint256 nomineeVotingPeriod;
    uint256 memberVotingPeriod;
    uint256 _fullWeightDuration;
```